framework/src/multiapps/MultiApp.C

```diff
@@ -120,6 +120,21 @@ MultiApp::MultiApp(const InputParameters & parameters):
 
   _total_num_apps = _positions.size();
   mooseAssert(_input_files.size() == 1 || _positions.size() == _input_files.size(), "Number of positions and input files are not the same!");
+
+  // Fill in the _positions vector
+  fillPositions();
+
+  if (_move_apps.size() != _move_positions.size())
+    mooseError("The number of apps to move and the positions to move them to must be the same for MultiApp " << name());
+
+  /// Set up our Comm and set the number of apps we're going to be working on
+  buildComm();
+
+  _backups.resize(_my_num_apps);
+
+  // Initialize the backups
+  for (unsigned int i=0; i<_my_num_apps; i++)
+    _backups[i] = MooseSharedPointer<Backup>(new Backup);
```

Comment:
Interesting, I believe this logic was originally in the constructor but I moved it all out when I fixed up all the multiapp initialization stuff. I don't see anything here that seems like it's in the wrong spot but this startup crap is really tricky...